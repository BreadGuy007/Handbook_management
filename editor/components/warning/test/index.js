/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import Warning from '../index';

describe( 'Warning', () => {
	it( 'should match snapshot', () => {
		const wrapper = shallow( <Warning>error</Warning> );

		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should has valid class', () => {
		const wrapper = shallow( <Warning /> );

		expect( wrapper.hasClass( 'editor-warning' ) ).toBe( true );
		expect( wrapper.find( '.editor-warning__actions' ) ).toHaveLength( 0 );
	} );

	it( 'should show child error message element', () => {
		const wrapper = shallow( <Warning actions={ <button /> }>Message</Warning> );

		const actions = wrapper.find( '.editor-warning__actions' );
		const action = actions.childAt( 0 );

		expect( actions ).toHaveLength( 1 );
		expect( action.hasClass( 'editor-warning__action' ) ).toBe( true );
		expect( action.childAt( 0 ).type() ).toBe( 'button' );
	} );
} );
