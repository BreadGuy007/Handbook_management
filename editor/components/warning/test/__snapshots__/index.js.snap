// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warning should match snapshot 1`] = `
<div
  className="editor-warning"
>
  <Dashicon
    icon="warning"
  />
  error
</div>
`;
